hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterImpl.java

```diff
@@ -548,6 +547,8 @@ private void doCacheOnWrite(long offset) {
       HFileBlock cacheFormatBlock = blockWriter.getBlockForCaching(cacheConf);
       cache.cacheBlock(new BlockCacheKey(name, offset, true, cacheFormatBlock.getBlockType()),
           cacheFormatBlock);
+      // refCnt will auto increase when block add to Cache, see RAMCache#putIfAbsent
+      cacheFormatBlock.release();
```

Comment:
Better to release the cacheFormatBlock in the finally block ?  Say even if we encounter some exception when cache#cacheBlock, the byte buffer allocated from ByteBuffAllocator can still have the chance to release ( return back to the pool ) ..